compiler/src/dotty/tools/dotc/typer/RefChecks.scala

```diff
@@ -782,7 +782,53 @@ object RefChecks {
         report.error(problem(), clazz.srcPos)
       }
 
+      // Checks base types of the current clazz against each other
+      //
+      // In particular, it checks that there are no two base classes with
+      // different type instantiations.
+      //
+      // ported from Scala 2:
+      //   https://github.com/scala/scala/blob/9bb659e62a9239c01aec14c171f8598bb1a576fe/src/compiler/scala/tools/nsc/typechecker/RefChecks.scala#L834-L883
```

Comment:
No need to put that in the code, pointing to it in the commit message is enough (otherwise we'd be linking to Scala 2 code everywhere).